Philips.CodeAnalysis.MaintainabilityAnalyzers/Readability/AvoidRedundantSwitchStatementAnalyzer.cs

```diff
@@ -12,25 +12,42 @@ namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Readability
 	[DiagnosticAnalyzer(LanguageNames.CSharp)]
 	public class AvoidRedundantSwitchStatementAnalyzer : DiagnosticAnalyzer
 	{
+		private readonly GeneratedCodeAnalysisFlags _generatedCodeFlags;
+
 		private const string Title = @"Do not use redundant switch statements";
 		private const string MessageFormat = @"Switch statement only has a default case.  Remove the switch statement and just use the default case code.";
 		private const string Description = @"Elide the switch statement";
 		private const string Category = Categories.Readability;
 
 		public static DiagnosticDescriptor Rule = new(Helper.ToDiagnosticId(DiagnosticIds.AvoidSwitchStatementsWithNoCases), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
 
+		public AvoidRedundantSwitchStatementAnalyzer()
+			: this(GeneratedCodeAnalysisFlags.None)
+		{ }
+
+		public AvoidRedundantSwitchStatementAnalyzer(GeneratedCodeAnalysisFlags generatedCodeFlags)
```

Comment:
This could also be done with one constructor and a default value for the parameter, right?

Reply:
Correct. Out of habit I spell them out to reduce chances of breaking public apis, which is semi-easy to do with the syntactic sugar of default values, but in this case, it makes good sense. Updated.

Reply:
See #176